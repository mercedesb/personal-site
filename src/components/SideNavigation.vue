<template>
  <div class='SideNav'>
    <ul class='SideNav-navLinks' v-if="navLinks.length">
      <li class="SideNav-navLink">
        <smart-link to="/">
          Home
        </smart-link>
      </li>
      <li class="SideNav-navLink" v-for="navLink in navLinks" :key="navLink.title">
        <smart-link
          :to="navLink.parsedLink"
          :isExternal="!!navLink.externalLink"
          >
          {{navLink.title}}
        </smart-link>
      </li>
    </ul>
    <div class="SideNav-background" v-if="backgroundImages && backgroundImages.length">
      <img
        v-for="image in backgroundImages"
        :key="image.id"
        :src="image"
        class="SideNav-backgroundImage"
      />
    </div>
  </div>
</template>

<script>
import images from '../mixins/images'

export default {
  mixins: [
    images
  ],
  props: {
    navLinks: {
      type: Object,
      required: true
    },
    backgroundImages: {
      type: Object
    }
  }
}
</script>

<style lang="scss">
@import '../assets/styles/variables.scss';

.SideNav {
  display: none;

  flex: 1;
  flex-direction: row-reverse;
  position: relative;
  width: 100%;
  height: auto;
  width: 300px;
  flex-direction: column;

  @include media($min-desktop) {
    display: flex;
  }

  &-navLinks {
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    z-index: 1;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }

  &-navLink {
    padding: $small-spacing 0;
    a {
      color: $white;
      font-size: $larger-font-size;
    }
  }

  &-background {
    display: none;
    flex: 1;
    align-items: center;
    flex-direction: column;

    @include media($min-tablet) {
      display: flex;
      justify-content: space-around;
    }
  }

  &-backgroundImage {
    max-width: 270px;
    min-width: 91px;
  }
}
</style>
