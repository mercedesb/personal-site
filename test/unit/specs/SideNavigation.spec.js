import SideNavigation from '@/components/SideNavigation.vue'
import { TestUtility } from './TestUtility'

describe('SideNavigation', () => {
  let component

  const initialProps = {
    backgroundImages: [
      {
        file: {
          url: 'http://placeholder.pics/svg/300'
        }
      },
      {
        file: {
          url: 'http://placeholder.pics/svg/300'
        }
      },
      {
        file: {
          url: 'http://placeholder.pics/svg/300'
        }
      }
    ],
    navLinks: Object.keys(TestUtility.landingPages).map((key) => {
      return TestUtility.landingPages[key]
    })
  }

  const shallow = propsData => TestUtility.shallow(SideNavigation, {
    propsData: {
      ...initialProps,
      ...propsData
    }
  })

  describe('Snapshots', () => {
    it('matches snapshot', () => {
      component = shallow()
      expect(component.element).toMatchSnapshot()
    })
  })

  describe('Properties', () => {
    beforeEach(() => {
      component = shallow()
    })

    it('has a navLinks property', () => {
      expect(component.props().navLinks).toBe(initialProps.navLinks)
    })

    it('has a backgroundImages property', () => {
      expect(component.props().backgroundImages).toBe(initialProps.backgroundImages)
    })
  })
})
