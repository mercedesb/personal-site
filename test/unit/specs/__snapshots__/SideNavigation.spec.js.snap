// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation Snapshots matches snapshot 1`] = `
<div
  class="SideNav"
>
  <ul
    class="SideNav-navLinks"
  >
    <li
      class="SideNav-navLink"
    >
      <smart-link-stub
        to="/"
      >
        
        Home
      
      </smart-link-stub>
    </li>
     
    <li
      class="SideNav-navLink"
    >
      <smart-link-stub>
        
        About
      
      </smart-link-stub>
    </li>
    <li
      class="SideNav-navLink"
    >
      <smart-link-stub>
        
        Blog
      
      </smart-link-stub>
    </li>
    <li
      class="SideNav-navLink"
    >
      <smart-link-stub>
        
        Connect
      
      </smart-link-stub>
    </li>
    <li
      class="SideNav-navLink"
    >
      <smart-link-stub
        isexternal="true"
      >
        
        Shop
      
      </smart-link-stub>
    </li>
  </ul>
   
  <div
    class="SideNav-background"
  >
    <img
      class="SideNav-backgroundImage"
      src="[object Object]"
    />
    <img
      class="SideNav-backgroundImage"
      src="[object Object]"
    />
    <img
      class="SideNav-backgroundImage"
      src="[object Object]"
    />
  </div>
</div>
`;
